src/main/kotlin/it/pagopa/wallet/services/PaymentMethodsService.kt

```diff
@@ -0,0 +1,70 @@
+package it.pagopa.wallet.services
+
+import it.pagopa.generated.ecommerce.model.PaymentMethodResponse
+import it.pagopa.wallet.client.EcommercePaymentMethodsClient
+import it.pagopa.wallet.repositories.PaymentMethod
+import it.pagopa.wallet.repositories.PaymentMethodsTemplateWrapper
+import kotlinx.coroutines.reactor.mono
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.switchIfEmpty
+
+@Service
+class PaymentMethodsService(
+    @Autowired private val paymentMethodsRedisTemplate: PaymentMethodsTemplateWrapper,
+    @Autowired private val ecommercePaymentMethodsClient: EcommercePaymentMethodsClient,
+) {
+
+    /*
+     * Logger instance
+     */
+    var logger: Logger = LoggerFactory.getLogger(this.javaClass)
+
+    fun getPaymentMethodById(paymentMethodId: String): Mono<PaymentMethodResponse> =
+        mono {
+                logger.info("Try to retrieve payment method from cache: [$paymentMethodId]")
```

Comment:
Maybe we can set some log to debug, at the moment we log 3 times every time we call this method

Reply:
Do you think it's appropriate to leave only one info log per call, for example cache hit/miss log?

Reply:
Here 2cd3ac2078fba4378f03bf53c0c02205a8c7c673 I leave info log only for cache hit/miss. 
Let me know if you think it is ok.